src/core/Table/filters/FilterButtonBar.tsx

```diff
@@ -60,10 +60,23 @@ export const FilterButtonBar = (props: FilterButtonBarProps) => {
   return (
     <div className={cx('iui-button-bar', className)} style={style}>
       {children}
-      <Button styleType='high-visibility' onClick={setFilter}>
+      <Button
+        styleType='high-visibility'
+        onClick={(e: React.MouseEvent) => {
+          e.stopPropagation();
```

Comment:
Really curious why clicking on a filter button (which is located in a dropdown, nowhere close to the actual table cell) would trigger sorting

Reply:
@basanta078 Could you investigate the root cause of this?

Reply:
even though filter button might look like it should not affect it, it is rendered for the same column and share the parent which is the column header

in react events propagate this way, popover's parent is the column header

Reply:
i moved the logic to base filter to prevent all clicks going to column header.
it sorted column even on clicking the text box
now it should fix all accidental clicks from the filter ui